### DISCUSSION TITLE
Implementing Matter in new environments example

### DISCUSSION THREAD

**Discussion by Orzech99** [matter-js/matter.js#1005] [Q&A]
## Overview
Hello!  I am currently working on integrating the core Matter.js functionality into a React Native application. From my understanding, there are several platform-specific parts that need to be implemented on my own. These are:

- BigInt/UInt8Array
- Network
- Crypto
- Storage
- Time
- BLE 
- Wi-Fi/Thread Commissioning

## Questions
1. Is my understanding correct that I need to implement these functionalities to get Matter.js working in a React Native environment?
2. Could you provide guidance or examples on where and how to implement these interfaces?

## Additional Context
I have successfully used matter-node.js in a Node.js environment and am now transitioning to React Native. Any advice or pointers to relevant documentation would be greatly appreciated.

## Future Contribution
After managing to get this to work, I plan to share my solution with the community. My side goal is to make the Matter protocol and Matter.js more accessible to a wider range of developers, especially those working with React Native.

**Thank you for your help!**

**Comment by Apollon77** [comment]
@Orzech99 Did you saw https://github.com/project-chip/matter.js/tree/main/packages/matter.js-react-native ?! We currently already work on react native and should be near to done ... The PR https://github.com/project-chip/matter.js/pull/988 is the current next/last step for next testing round.

So if you are familiar with react native please have a look at the package and code (there also your questions should be answered) also on the PR and provide Feedback if you see anything strange or broken.

With this last PR we "should" be fine because anything should be integrated ... reality tests will then show where ewe really stand. If you like to join these experiments and support in finalization you're welcome :-)

I hope you did not do too much work already, but if then maybe a good idea to compare :-)

**Reply by Apollon77** [reply]
I changed it into a discussion.

Additionally, re "Wi-Fi/Thread Commissioning" here we should talk what exactly you mean. I would assume that the mobile device yuo use already has wifi connected and Thread is a question ... so here for "device" side it is more about what's the plan and we might need to discuss requirements

**Reply by Orzech99** [reply]
I did not see either PR nor the package.
I'll check it out and see if I can make it work.

Regarding "Wi-Fi/Thread Commissioning" just forget about it, I was really tired and you could say it appeared there by mistake, I apologise for the confusion.

**Reply by Apollon77** [reply]
PR see link in my first post.
The packages itself afte rcurrently just in the nightly builds. user @dev npm tag to install these

**Reply by Orzech99** [reply]
I did see that, I pulled the latest update and built it on my own, now using it as a file package dependency so in case something 'goes wrong' I can look at the build package and trace back without searching in node_modules.

**Reply by Orzech99** [reply]
In my package.json:
```
"matter.js-react-native": "file:./packages/matter.js-react-native",
```

**Reply by joey01245** [reply]
Is there any update on this? @Orzech99 Did you figure this out?

**Reply by Orzech99** [reply]
@joey01245 so … not everything that needs to be implemented for matter.js-react-native is implemented yet, we (@Apollon77 and me) did a couple more steps forward to make matter.js work on RN and got further but I decided to learn swift and kotlin to make native apps because matter sdk exists for both and my project has a strict timeline. If you are down to push this forward join the matter.js discord and contact @Apollon77 maybe you could make the final changes and get it to work on RN. My time is limited but I’m always happy to help - if you get somewhere and need help message me on discord as well. Apollon77 can connect us through discord.

**Reply by Apollon77** [reply]
In fact I am no mobile developer, so i fact if someone shares logs and errors I can solve issues and kill road blocks, but I'm currently not able do this myself. So if someone wants to use the package we have and continue the testing I am happy to support.
